examples/basic/ui/selection_custom_toolbar.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+
+from bokeh.io import show
+from bokeh.models import BoxSelectTool, Button, CustomJS, PaletteSelect, Panel, Row
+from bokeh.palettes import Spectral11
+from bokeh.plotting import figure
+
+
+def plot(N: int):
+    x = np.random.random(size=N) * 100
+    y = np.random.random(size=N) * 100
+    radii = np.random.random(size=N) * 1.5
+    colors = np.random.choice(Spectral11, size=N)
+
+    p = figure(active_scroll="wheel_zoom", lod_threshold=None, title=f"Plot with N={N} circles")
+    cr = p.circle(x, y, radius=radii, fill_color=colors, fill_alpha=0.6, line_color=None)
+
+    return p, cr
+
+p, cr = plot(500)
+
+box_select = BoxSelectTool(persistent=True)
+p.add_tools(box_select)
+
+common = dict(margin=0, sizing_mode="stretch_height")
+
+delete = Button(label="Delete", **common)
+select = PaletteSelect(
+    value=Spectral11[0],
+    items=[(color, [color]) for color in Spectral11],
+    swatch_width=30,
+    **common,
+)
+clear = Button(label="Clear", **common)
+
+toolbar = Panel(
+    position=box_select.overlay.nodes.bottom_left,
```

Comment:
Where can I read about the `nodes` object? Does this resolve on render or does `nodes.bottom_left` represent some actual concrete value?

Reply:
I need to write documentation for nodes. It resolves during geometry/positioning computations, so still technically on render/paint. I'm in the process of separating those two phases, but it won't be done here. Computed values are only available in bokehjs. It may be possible to expose those values in a similar way to how computed layouts are exposed, e.g. in plots.

Reply:
I think syncing it is probably overkill, was just trying to understand how it works.

Reply:
I considered adding support for reporting computed values. For performance reasons, that would be done via a subscription based mechanism similar to how JS event callbacks work.